### DISCUSSION TITLE
How to label the points on the graph itself using plot_gsva_pca()?

### DISCUSSION THREAD

**Discussion by levinhein** [reactome/ReactomeGSA#31] [Q&A] [answered]
What is the parameter in order to label the points inside the graph itself? It's hard to distinguish colors when there are too many data points. Adding "label = TRUE" seems not to be the right way to do it.  

**`plot_gsva_pca(gsva_result)`**

![image](https://user-images.githubusercontent.com/68268222/108681651-5defb980-752a-11eb-913c-19c75619eb00.png)

**Comment by jgriss** [comment, marked as answer]
Hi @levinhein,

Thanks a lot for your interest in ReactomeGSA!

You can do this quite easily using standard `ggplot2` functions. The following code adds labels on top of the points:

```R
# After you received your GSVA result
plot_gsva_pca(gsva_result) + 
    geom_text(aes(label=sample), 
         nudge_x = 0.005, 
         nudge_y = 0.005, 
         check_overlap = T)
```

You'll probably have to adapt `nudge_x` and `nudge_y` to your result since it's based on the arbitrary PCA units.

Kind regards,
Johannes
